featurebyte/worker/__init__.py

```diff
@@ -8,47 +8,76 @@
 from featurebyte.utils.messaging import REDIS_URI
 from featurebyte.utils.persistent import DATABASE_NAME, MONGO_URI
 
-CELERY_TASK_COLLECTION = Task.collection_name()
-CELERY_SCHEDULE_COLLECTION = PeriodicTask.collection_name()
-
-celery = Celery(
-    __name__,
-    broker=REDIS_URI,
-    backend=MONGO_URI,
-    task_serializer="json",
-    accept_content=["json"],
-    result_serializer="json",
-    enable_utc=True,
-)
-
-# Celery configuration options:
-# https://docs.celeryq.dev/en/stable/userguide/configuration.html
-
-# celery behavior
-celery.conf.task_track_started = True
-
-# task queues and routing
-celery.conf.task_routes = {
-    "featurebyte.worker.task_executor.execute_cpu_task": {"queue": "cpu_task"},
-    "featurebyte.worker.task_executor.execute_io_task": {"queue": "io_task"},
-}
-
-celery.conf.broker_transport_options = {
-    "priority_steps": list(range(3)),
-    "sep": ":",
-    "queue_order_strategy": "priority",
-}
-
-# task tombstones options
-celery.conf.result_extended = True
-celery.conf.result_expires = 15  # 15 days expiry
-celery.conf.result_backend_always_retry = False
-celery.conf.mongodb_backend_settings = {
-    "database": DATABASE_NAME,
-    "taskmeta_collection": CELERY_TASK_COLLECTION,
-}
-
-# beat scheduler
-celery.conf.mongodb_scheduler_db = DATABASE_NAME
-celery.conf.mongodb_scheduler_collection = CELERY_SCHEDULE_COLLECTION
-celery.conf.mongodb_scheduler_url = MONGO_URI
+CELERY = None
+
+
+def get_celery(
+    redis_uri: str = REDIS_URI, mongo_uri: str = MONGO_URI, database_name: str = DATABASE_NAME
+) -> Celery:
+    """
+    Get Celery instance
+
+    Parameters
+    ----------
+    redis_uri: str
+        Redis URI
+    mongo_uri: str
+        Mongo URI
+    database_name: str
+        Database name
+
+    Returns
+    -------
+    Celery
+    """
+    global CELERY
+    if CELERY:
+        return CELERY
```

Comment:
Do we need to consider whether this is thread-safe or not?

Reply:
Noted that it is not related to the current refactoring PR. Just check whether we need to fix this. If the construction of the celery object is light, would prefer to make it like the `get_persistent` and don't store the global object. It is also more future proof in case someone in the future introduce some states to the celery instance.

Reply:
There is some cost to calling the function each time, but it's not very large
```
>>> timeit.timeit("get_celery()", globals=globals(), number=1000)
0.046812791999997216
>>> timeit.timeit("celery", globals=globals(), number=1000)
4.016700000875062e-05
```

Reply:
Seems like it is costing 4ms on average. We can improve this later if it becomes a bottleneck.